app/Http/Controllers/API/Exports/External/CliController.php

```diff
@@ -26,29 +26,35 @@ protected function getResource()
         return 'export_jobs';
     }
 
+	protected function getUsecase()
+	{
+		if (!$this->usecase) {
+			// @todo inject
+			$this->usecase = service('factory.usecase')
+				->get('posts_export', 'export');
+		}
+		return $this->usecase;
+	}
     public function show(Request $request)
     {
-    	$route_params = $this->getRouteParams($request);
-        // this is a trick to convert 'false' to falsy (which would be true),
-        // 'true' to true, and an unset param to false
-        $include_header = json_decode($request->input('include_header', 1)) == true ? 1 : 0;
-		// Run export command
-        $exitCode = Artisan::call('export', [
-			'job' => $route_params['id'],
-            '--limit' => $request->input('limit', 0),
-            '--offset' => $request->input('offset', 0),
-            '--include-header' => $include_header,
-        ]);
-
-        // Retrieve the results of rhe export
-        // which should be a json formatted string
-        // containing information aboutt he file generated and
-        // saved by the exporter
-        $output = Artisan::output();
-        $file_details = json_decode($output, true);
-
-        return $this->prepResponse([
-            'results' => $file_details,
-        ], $request);
+		$route_params = $this->getRouteParams($request);
+		// this is a trick to convert 'false' to falsy (which would be true),
+		// 'true' to true, and an unset param to false
+		$include_header = json_decode($request->input('include_header', 1)) == true ? 1 : 0;
+
+		// set CLI params to be the payload for the usecase
+		$payload = [
+			'job_id' => $route_params['id'],
+			'limit' => $request->input('limit', 0),
+			'offset' => $request->input('offset', 0),
+			'add_header' => $include_header
+		];
+		// Get the usecase and pass in authorizer, payload and transformer
+		$this->usecase = $this->getUsecase()
+			->setPayload($payload)
```

Comment:
It seems like these should be filters/search params as they normally are in a search usecase